packages/@webex/plugin-cc/test/unit/spec/AgentConfig/AgentConfig.ts

```diff
@@ -0,0 +1,276 @@
+import { WebexSDK } from '../../../../src/types';
+import { WORK_TYPE_CODE } from '../../../../src/AgentConfig/types';
+import AgentConfig from '../../../../src/AgentConfig/AgentConfig';
+import {
+  DesktopProfileResponse,
+  ListAuxCodesResponse,
+  Team,
+  AgentResponse,
+} from '../../../../src/AgentConfigService/types';
+
+const mockWebex: WebexSDK = {
+  internal: {
+    device: {
+      orgId: 'testOrgId',
+    },
+  },
+};
+
+const mockAgentResponse: AgentResponse = {
+  firstName: 'John',
+  lastName: 'Doe',
+  agentProfileId: 'profile123',
+  email: 'john.doe@example.com',
+  teamIds: ['team1', 'team2'],
+};
+
+const mockDesktopProfileResponse: DesktopProfileResponse = {
+  loginVoiceOptions: ['option1', 'option2'],
+  accessWrapUpCode: 'ALL',
+  accessIdleCode: 'ALL',
+  wrapUpCodes: [],
+  idleCodes: [],
+};
+
+const mockTeamsListResponse: Team[] = [{ id: '123', name: 'Team 1' }];
+
+const mockAuxCodesResponse: ListAuxCodesResponse = {
+  data: [
+    { id: 'aux1', active: true, defaultCode: true, isSystemCode: true, description: 'test', name: 'testName', workTypeCode: WORK_TYPE_CODE.WRAP_UP_CODE,},
+    { id: 'aux1', active: true, defaultCode: true, isSystemCode: true, description: 'test', name: 'testName', workTypeCode: WORK_TYPE_CODE.IDLE_CODE,}
+  ],
+};
+
+describe('AgentConfig', () => {
+  let agentConfig: AgentConfig;
+  let getUserUsingCISpy: jest.SpyInstance;
+  let getDesktopProfileByIdSpy: jest.SpyInstance;
+  let getListOfTeamsSpy: jest.SpyInstance;
+  let getListOfAuxCodesSpy: jest.SpyInstance;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    agentConfig = new AgentConfig('agent123', mockWebex, 'http://api.url');
+
+    getUserUsingCISpy = jest.spyOn(AgentConfigService.prototype, 'getUserUsingCI').mockResolvedValue(mockAgentResponse);
+    getDesktopProfileByIdSpy = jest.spyOn(AgentConfigService.prototype, 'getDesktopProfileById').mockResolvedValue(mockDesktopProfileResponse);
+    getListOfAuxCodesSpy = jest.spyOn(AgentConfigService.prototype, 'getListOfAuxCodes').mockResolvedValue(mockAuxCodesResponse);
+    getListOfTeamsSpy = jest.spyOn(AgentConfigService.prototype, 'getListOfTeams').mockResolvedValue(mockTeamsListResponse);
+    
+  });
+
+  it('should fetch agent profile successfully', async () => {
+    const expectedProfile: IAgentProfile = {
+      agentId: 'agent123',
+      name: 'John Doe',
+      agentProfileId: 'profile123',
+      agentMailId: 'john.doe@example.com',
+      teams: mockTeamsListResponse,
+      loginVoiceOptions: ['option1', 'option2'],
+      wrapUpCodes: [{id: 'aux1', active: true, defaultCode: true, isSystemCode: true, description: 'test', name: 'testName', workTypeCode: WORK_TYPE_CODE.WRAP_UP_CODE}],
+      idleCodes: [{id: 'aux1', active: true, defaultCode: true, isSystemCode: true, description: 'test', name: 'testName', workTypeCode: WORK_TYPE_CODE.IDLE_CODE}]
+    };
+
+    expect(getUserUsingCISpy).toHaveBeenCalledOnceWith;
```

Comment:
How is this passing? This is not correct. We need to pass expected value in toHaveBeenCalledOnceWith(). And these expect statements should be after getAgentProfile is invoked

Reply:
Discussed over Webex & made changes according to this.